pyproject.toml

```diff
@@ -24,6 +43,57 @@ line-length = 120
 exclude = '(_notebooks/.*)'
 
 
+[tool.ruff]
+line-length = 120
+# Enable Pyflakes `E` and `F` codes by default.
+select = [
+    "E", "W",  # see: https://pypi.org/project/pycodestyle
+    "F",  # see: https://pypi.org/project/pyflakes
+# TODO
+#    "D",  # see: https://pypi.org/project/pydocstyle
+#    "N",  # see: https://pypi.org/project/pep8-naming
+]
+#extend-select = [
+#    "C4",  # see: https://pypi.org/project/flake8-comprehensions
+#    "PT",  # see: https://pypi.org/project/flake8-pytest-style
+#    "RET",  # see: https://pypi.org/project/flake8-return
+#    "SIM",  # see: https://pypi.org/project/flake8-simplify
+#]
```

Comment:
How did you choose these? We didn't have them before

Reply:
yes, we did not that is why they are commented but would experiment with then in the following work...

Reply:
Then I suggest you remove these from this PR and add them separately on those followups